wsd/Storage.hpp

```diff
@@ -239,16 +192,14 @@ class StorageBase
     class UploadResult final
     {
     public:
-        enum class Result
-        {
-            OK = 0,
-            DISKFULL,
-            TOO_LARGE, //< 413
-            UNAUTHORIZED, //< 401, 403, 404
-            DOC_CHANGED, /**< Document changed in storage */
-            CONFLICT, //< 409
-            FAILED
-        };
+        STATE_ENUM(Result,
+                   OK = 0, //< Uploaded successfully
+                   DISKFULL, //< Unused.
+                   TOO_LARGE, //< 413
+                   UNAUTHORIZED, //< 401, 403, 404
+                   DOC_CHANGED, /**< Document changed in storage */
+                   CONFLICT, //< 409
+                   FAILED);
```

Comment:
STATE_ENUM produces an int8_t enum class, OK I guess. Would use uint8_t at least though (actually uint32_t anyways due to min-arch probably is 32-bit).

STATE_ENUM also takes care for a 'const char*' conversion, nice. We use a static local immutable string array - nice + the mandatory toString(..). Nice.